src/main/java/com/google/cloud/tools/appengine/cloudsdk/CloudSdk.java

```diff
@@ -651,6 +651,14 @@ public Builder inheritProcessOutput(boolean inheritProcessOutput) {
     }
 
     /**
+     * Sets the desired Java SDK path, used in devappserver runs and App Engine standard staging.
+     */
+    public Builder javaSdkPath(Path javaSdkPath) {
```

Comment:
This is an odd name as there is no Java SDK: there's a JRE and a JDK, and JDKs usually include the JRE in `jre/`.  `javaHome()` or `javaInstallRoot()`?

<hr>
I hesitate in suggesting the following…

It would be more flexible if we could instead provide a JVM runner/launcher object.  In the Eclipse JDT world, the user can configure multiple JREs, each of which can have additional VM arguments and libraries to form the bootclasspath, and there's already infrastructure for auto-finding JVMs and determining these flags.
Perhaps something like:
```
interface JvmRunner {
   void validate() throws AppEngineException;
   void run(String mainClass, String[] args, String[] jvmArgs, String[] classpath);
}
```
and move much of the changes above into a implementation `StandardJavaRunner`?  Which we would ignore and instead create a wrapper around the JDT `IVMRunner`.

My concern is that what we execute may not match what the user has configured in their JRE description